modules/javafx.graphics/src/main/native-glass/win/Robot.cpp

```diff
@@ -176,13 +178,14 @@ JNIEXPORT void JNICALL Java_com_sun_glass_ui_win_WinRobot__1mousePress
 
     // Support for extra buttons
     if (buttons & (1 << 3)) {
-        mouseInput.mi.dwFlags |= MOUSEEVENTF_XDOWN;
-        mouseInput.mi.mouseData = XBUTTON1;
+        dwFlags |= MOUSEEVENTF_XDOWN;
+        mouseFlags |= XBUTTON1;
     }
     if (buttons & (1 << 4)) {
-        mouseInput.mi.dwFlags |= MOUSEEVENTF_XDOWN;
-        mouseInput.mi.mouseData = XBUTTON2;
+        dwFlags |= MOUSEEVENTF_XDOWN;
+        mouseFlags |= XBUTTON2;
     }
```

Comment:
The above two blocks need to be moved before the setting of the mouseInput structure (else the dwFlags setting will not have any effect).

Reply:
Fixed.